### PR TITLE
Memory Devices improvements

### PR DESCRIPTION
**Pull request by bp0**
* note if ddr3 is installed but ee1004 is used
* create an SPD "array" if no DMI arrays exist
* show mem types in array info
* include any SPD-detected modules even if they are
  types that can't be decoded currently

And more:
* DMI: +Rank
* SPD: +DDR3 mfg_date, ranks, iopins, voltages, dram vendor
* bug fixes

SPD only:
Screenshot from 2019-06-28 12-24-33

DMI+SPD:
Screenshot from 2019-06-28 12-26-09

### CODE REVIEW

File: hardinfo/dmi_util.c [hunk 1]
```diff
@@ -234,6 +234,11 @@ char *dmi_chassis_type_str(int chassis_type, gboolean with_val) {
     return NULL;
 }
 
+/* TODO: something better maybe */
+char *dd_cache[128] = {};
```
Comment (lpereira) [inline hardinfo/dmi_util.c]: Maybe `static`?

**Comment by bp0** [comment]
Computer Summary shows size and type under RAM if DMI or SPD information is available. otherwise falls back to using /proc/meminfo source as before.

RAM Screenshot_2019-06-28_23-07-06

File: hardinfo/dmi_util.c (hunk 1 above)
Comment (bp0) [reply hardinfo/dmi_util.c]: Indeed.

**Comment by bp0** [comment]
Ok, rebase-master-ed. I think I'm done with memory devices for now. @lpereira, please merge.

**Comment by lpereira** [comment]
Great work on this, @bp0!

**Pull request merged**